rfcs/debuggability_metrics

```diff
@@ -0,0 +1,126 @@
+
+There is also a wide range of improvements that can be done for debugability timelines that cross over into other areas. For example, faster build times also improves debugability timelines. However, to limit the scope of this RFC, we are going to limit this document to discussing elements that are unique to debugability.
+
+We are suggesting to measure a few key debugging times and polling customers on their debugging experience.
+
+## What is the motivation for this suggestion?
+
+The primary is to enable metrics-driven decisions around improvements to debugability. Diving a little deeper, these metrics will allow us to more intelligently focus our efforts in improving debugability, specifically in improving the timelines for discovering and fixing bugs. It will also allow us to prioritize which problems are most severe.
+
+We also want to be able to track the impact of decisions we make to improve debugability. Measuring metrics and their subsequent improvement is one way to do this. This will allow us to provide visibility to the larger organization by a metrics-driven approach and provide support for claims of debugability improvements.
+
+## Design Description
+
+### Debugging Steps
+
+Here are some common steps of finding and fixing a bug:
+
+1. The error is reported. This is done by Jenkins sending out an e-mail, although developers can also view the Jenkins build through its web UI.
+2. The log is investigated. This can be found through the web UI for Jenkins. (or in a GitHub issue)
+   * This may include artifact investigation, which is also found through the web UI for Jenkins
+3. The bug is reproduced locally. This step is not always required, but often is. Depending on the developer's workstation state, the nature of the bug, and the developer's preferences, the following sub-steps may be required:
+   1. Sync the commit where the bug was found.
+   2. Build the commit in the appropriate flavor.
+   3. Add breakpoints. This is a less common step.
+   4. Add new output. This is a less common step.
+   5. Run the appropriate executable(s).
+   6. Run the appropriate test(s).
+4. Make changes to actually fix the bug.
+   * If no test existed to catch the bug, also add an appropriate test(s) to prevent the bug from regressing.
+
+### Measuring Debugging Time
+
+Each of the steps above can be measured in time, although the value of those data points will vary wildly. Many of them are too dependent on the bug, can be done in parallel, or are out of scope for debugability. In order to ensure consistent measuring, we will perform a UX study where research subjects are monitored by a researcher. The researcher will measure the time each step takes with a stopwatch, record the results, and collect other feedback.
```

Comment:
Setting up extensive UX experiments with time points seems like overkill, especially when developers can directly propose changes/issues for negative debugging experience improvements. Suggest metrics which aren't automatable are proactively collected only through user surveys.

This would simplify this proposal to:
1. Identify short-term automatable debuggability/issue-lifecycle metrics
2. Identify questions for a developer survey